### PR TITLE
Adding the backend for Loans

### PR DESCRIPTION
**Pull request by Patrick-Pataky**
The LoanScreen has now a proper LoanViewModel, fully tested.

### CODE REVIEW

File: app/src/main/java/com/android/partagix/model/filtering/Filtering.kt
Comment (skyrag) [inline app/src/main/java/com/android/partagix/model/filtering/Filtering.kt:1]: in the function filterItems where we need the radius, the list parameter is missing in the javadoc

Comment (skyrag) [inline app/src/main/java/com/android/partagix/model/filtering/Filtering.kt:1]: the last funtion is missing the description for the list parameter in the javadoc

File: app/src/main/java/com/android/partagix/model/LoanViewModel.kt
Comment (skyrag) [inline app/src/main/java/com/android/partagix/model/LoanViewModel.kt:1]: line44 : TO-DO left

**Review by skyrag — CHANGES_REQUESTED** [review]
one or two things to change

File: app/src/main/java/com/android/partagix/model/LoanViewModel.kt
Comment (Patrick-Pataky) [reply app/src/main/java/com/android/partagix/model/LoanViewModel.kt:1]: A mechanism that needs to display a message showing that the user is not connected must be first implemented. That's why the TODO is kept.

**Pull request merged**